File: MySocialMedia/src/main/java/com/example/MySocialMedia/services/FeedService.java
package com.example.MySocialMedia.services;

import com.example.MySocialMedia.baens.Following;
import com.example.MySocialMedia.baens.Post;
import com.example.MySocialMedia.baens.PostComment;
import com.example.MySocialMedia.baens.User;
import com.example.MySocialMedia.repositories.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

@Service
public class FeedService {
    @Autowired
    private PostRepository postRepository;
    @Autowired
    private FollowingRepo followingRepo;
    @Autowired
    private PostCommentRepo postCommentRepo;
    @Autowired
    private UserRepository userRepository;
    @Autowired
    private PostLikeRepo postLikeRepo;


    public List<Post> getAllPosts() {
        return postRepository.findAllByOrderByIdDesc();
    }

    public List<PostComment> postComments(long postId) {
        ;
        return postCommentRepo.findByPostIdOrderByNewest(postId);
    }

    public int getPostCommentsCount(long postId) {
        return postCommentRepo.getCommentCountByPostId(postId);
    }

    public Post getPost(long postId) throws SQLException {
        return postRepository.findById(postId).orElseThrow(() -> new SQLException("Post Not Found!"));
    }

    public User getUserDetails(long id) throws SQLException {
        return userRepository.findById(id).orElseThrow(() -> new SQLException("User By ID not Found"));
    }

    public List<Post> getPostsByUserId(long id) {
        return postRepository.findByAuthorIdOrdered(id);
    }

    public List<Post> getPostsByFollowing(User user) {
        List<Post> posts = new ArrayList<>();
        List<Post> temp;
        for (Following f : followingRepo.findByFollower_Id(user.getId())) {
            temp = postRepository.findByAuthor_Id(f.getFollowed().getId());
            posts.addAll(temp);
        }
        return posts;
    }

    public List<User> getUserFollowers(long userId) {
        return followingRepo.getAllFollowers(userId);
    }

    public List<User> getUserFollowings(long userId) {
        return followingRepo.getAllFollowings(userId);
    }

    public List<User> findAllPostLikers(long postId){
        return postLikeRepo.findUsersByPostId(postId);
    }
}

File: social_media_app/src/services/FeedService.ts
import axios from "axios";
import { Post } from "../models/Post";
import { PostComment } from "../models/PostComment";
import { User } from "../models/User";


class FeedService{
    async getAllPosts(){
        return (await axios.get<Post[]>(`http://localhost:8080/api/feed/posts`)).data
    }
    async getPostComments(postId:number){
        return (await axios.get<PostComment[]>(`http://localhost:8080/api/feed/postcomments/${postId}`)).data

    }
    async getPost(postId:number){
        return (await axios.get<Post>(`http://localhost:8080/api/feed/getpost/${postId}`)).data

    }

    async getPostCommentsCount(postId:number){
        return (await axios.get<number>(`http://localhost:8080/api/feed/getCommentsCount/${postId}`)).data

    }

    async getUserById(userId:number){
        return (await axios.get<User>(`http://localhost:8080/api/feed/getuserbyid/${userId}`)).data

    }

    async getUsersPostsById(userId:number){
        return (await axios.get<Post[]>(`http://localhost:8080/api/feed/getusersposts/${userId}`)).data
    }

    async getAllFollwers(userId:number){
        return (await axios.get<User[]>(`http://localhost:8080/api/feed/getuserfollowers/${userId}`)).data
      }
    
      async getAllFollwings(userId:number){
        return (await axios.get<User[]>(`http://localhost:8080/api/feed/getuserfollowings/${userId}`)).data
      }

      async findAllPostLikers(postId:number){
        return (await axios.get<User[]>(`http://localhost:8080/api/feed/findallpostlikers/${postId}`)).data
      }
  
}

const feedService = new FeedService;
export default feedService;
